### PR TITLE
Retire .NET API analyzer content as it is deprecated

### LINKED ISSUE #22856: Microsoft.DotNet.Analyzers.Compatibility is deprecated
The NuGet package Microsoft.DotNet.Analyzers.Compatibility is deprecated as indicated [here]( https://github.com/dotnet/platform-compat#platform-compatibility-analyzer). It is confusing that there are instructions to add this deprecated package to your project.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 3cf0043f-baf5-d2ab-84a5-718f900c0e25
* Version Independent ID: c3fabde5-775f-63f9-6b22-425435e096d8
* Content: [.NET API analyzer](https://docs.microsoft.com/en-us/dotnet/standard/analyzers/api-analyzer)
* Content Source: [docs/standard/analyzers/api-analyzer.md](https://github.com/dotnet/docs/blob/master/docs/standard/analyzers/api-analyzer.md)
* Product: **dotnet-fundamentals**
* Technology: **dotnet-analyzers**
* GitHub Login: @OliaG
* Microsoft Alias: **dapine**

### PR DESCRIPTION
**Pull request by IEvangelist**
## Summary

- Retire .NET API analyzer content as it is deprecated
- Adjust TOC accordingly - remove it from listing
- Apply appropriate redirect entry, now redirects to _platform-compat-analyzer.md_
- Update existing article refs and content

Fixes #22856

### CODE REVIEW

File: docs/standard/analyzers/portability-analyzer.md [hunk 1]
```diff
@@ -1,14 +1,15 @@
...
 
-Once you've converted your project to target the new platform, like .NET Core, you can use the Roslyn-based [API Analyzer tool](api-analyzer.md) to identify APIs throwing <xref:System.PlatformNotSupportedException> exceptions and other compatibility issues.
+Once you've converted your project to target the new platform, like .NET Core, you can use the Roslyn-based [Platform compatibility analyzer](platform-compat-analyzer.md) to identify APIs throwing <xref:System.PlatformNotSupportedException> exceptions and other compatibility issues.
```
Comment (gewarren) [inline docs/standard/analyzers/portability-analyzer.md:12]:
```suggestion
Once you've converted your project to target the new platform, like .NET Core, you can use the Roslyn-based [Platform compatibility analyzer](platform-compat-analyzer.md) to identify APIs that throw <xref:System.PlatformNotSupportedException> exceptions and other compatibility issues.
```

**Review by gewarren — APPROVED** [review]
I assume we don't need to archive the old material?

**Comment by IEvangelist** [comment]
> I assume we don't need to archive the old material?

What old material are you specifically referring to, I'm not really sure?

**Comment by gewarren** [comment]
> > I assume we don't need to archive the old material?
> 
> What old material are you specifically referring to, I'm not really sure?

This file: docs/standard/analyzers/api-analyzer.md

**Comment by IEvangelist** [comment]
> > > I assume we don't need to archive the old material?
> > 
> > 
> > What old material are you specifically referring to, I'm not really sure?
> 
> This file: docs/standard/analyzers/api-analyzer.md

I'm now realizing that "archive" is probably something more formal that I'm familiar with. I just assumed we'd delete it and redirect. Do you have any details on archiving, and how does one formally do that?

**Comment by gewarren** [comment]
"Archiving" means moving it to docs.microsoft.com/previous-versions. There's an Archive tool on the OPS portal that generates a PR to add it for you.

**Pull request merged**